algopy_testing/src/algopy_testing/contexts/state_context.py

```diff
@@ -0,0 +1,44 @@
+import threading
+from collections.abc import Generator
+from contextlib import contextmanager
+from typing import Any, Generic, TypeVar
+
+T = TypeVar("T")
+
+
+class TestContext(Generic[T]):
```

Comment:
I think the contextvar module would be better here https://docs.python.org/3/library/contextvars.html

Reply:
Let me know what you think on the current approach in latest commit